include/pops/simulation.hpp

```diff
@@ -247,80 +261,128 @@ class Simulation
     unsigned movement(
         IntegerRaster& infected,
         IntegerRaster& susceptible,
-        IntegerRaster& mortality_tracker,
+        std::vector<IntegerRaster>& mortality_tracker_vector,
+        std::vector<IntegerRaster>& exposed,
+        IntegerRaster& resistant,
         IntegerRaster& total_hosts,
+        IntegerRaster& total_exposed,
         unsigned step,
         unsigned last_index,
         const std::vector<std::vector<int>>& movements,
-        std::vector<unsigned> movement_schedule)
+        std::vector<unsigned> movement_schedule,
+        std::vector<std::vector<int>>& suitable_cells)
     {
-        UNUSED(mortality_tracker);  // Mortality is not supported by movements.
         for (unsigned i = last_index; i < movements.size(); i++) {
             auto moved = movements[i];
             unsigned move_schedule = movement_schedule[i];
             if (move_schedule != step) {
                 return i;
             }
             int infected_moved = 0;
+            int exposed_moved = 0;
             int susceptible_moved = 0;
+            int resistant_moved = 0;
             int total_hosts_moved = 0;
-            double inf_ratio = 0;
             int row_from = moved[0];
             int col_from = moved[1];
             int row_to = moved[2];
             int col_to = moved[3];
             int hosts = moved[4];
+            std::vector<int> exposed_categories;
+            std::vector<int> mortality_categories;
             if (hosts > total_hosts(row_from, col_from)) {
                 total_hosts_moved = total_hosts(row_from, col_from);
             }
             else {
                 total_hosts_moved = hosts;
             }
-            if (infected(row_from, col_from) > 0
-                && susceptible(row_from, col_from) > 0) {
-                inf_ratio = double(infected(row_from, col_from))
-                            / double(total_hosts(row_from, col_from));
-                int infected_mean = total_hosts_moved * inf_ratio;
-                if (infected_mean > 0) {
-                    if (movement_stochasticity_) {
-                        std::poisson_distribution<int> distribution(infected_mean);
-                        infected_moved = distribution(generator_);
-                    }
-                    else {
-                        infected_moved = infected_mean;
-                    }
+            auto total_infecteds = infected(row_from, col_from);
+            auto suscepts = susceptible(row_from, col_from);
+            auto expose = total_exposed(row_from, col_from);
```

Comment:
I don't remember whether you actually want to add total_exposed in general or it is just for this. If you put the `for (auto& raster : exposed) {` loop into forming of categories starting the index at 4, you won't need total_exposed. You would not have (total_)infected here in categories, but you would loop through exposed after the draw to get the (total_)infected. However, if you want total_exposed, then the current code is actually better then what I'm suggesting.

Reply:
Other than using it here, it isn't useful for the model at this point but could be very useful for visualization purposes on the dashboard. I was planning on calculating it in the interface but now it can just be exported saving multiple loops in the interface. A long way to say that I think it is worth keeping

Reply:
Sounds good!